gsy_framework/community_datasheet/community_datasheet_parser.py

```diff
@@ -0,0 +1,182 @@
+                    location_converter=location_converter,
+                    session=session)
+                for pv_asset in pv_assets:
+                    pv_asset["geo_tag_location"] = coordinates
+
+    @staticmethod
+    def _get_member_coordinates(
+            member_details: Dict, location_converter: LocationConverter, session=None):
+        address = member_details["Location/Address (optional)"] or ""
+        zip_code = member_details["ZIP code"] or ""
+        full_address = f"{address} {zip_code}"
+        if not full_address.strip():
+            return None
+
+        try:
+            return location_converter.convert(full_address, session=session)
+        except LocationConverterException as ex:
+            raise CommunityDatasheetException(ex) from ex
+
+    @staticmethod
+    def _validate_grid(grid):
+        try:
+            scenario_validator(grid)
+        except ValidationError as ex:
+            raise CommunityDatasheetException(ex) from ex
+
+    def _get_assets_by_member(self) -> Dict:
+        """Return a mapping between each member and their assets."""
+        assets_by_member = {member_name: [] for member_name in self._datasheet.members}
+        asset_items = [
+            assets.items()
+            for assets in [self._datasheet.loads, self._datasheet.pvs, self._datasheet.storages]
+        ]
+        for member_name, assets in chain.from_iterable(asset_items):
+            assets_by_member[member_name].extend(assets)
+
+        if any((missing := member) not in self._datasheet.members for member in assets_by_member):
+            raise CommunityDatasheetException(
+                f'Member "{missing}" was defined in one of the asset sheets'  # noqa: F821
+                'but not in the "Community Members" sheet.')
+
+        return assets_by_member
+
+    @staticmethod
+    def _merge_profiles_into_assets(profiles: Dict, assets_by_member: Dict) -> None:
+        """Merge (in-place) each energy profile into the representation of its own asset."""
+        for asset in chain.from_iterable(assets_by_member.values()):
+            asset_name = asset["name"]
+            if asset_name not in profiles:
+                continue
+            asset_type = asset["type"]
+            try:
+                profile_key = PROFILE_KEYS_BY_TYPE[asset_type]
+            except KeyError as ex:
+                raise CommunityDatasheetException(
+                    f'Asset type "{asset_type}" is not supported.') from ex
+            asset[profile_key] = profiles[asset_name]
+
+    def _create_grid(self, assets_by_member: Dict) -> Dict:
+        members_grid = []
+        for member_name, assets in assets_by_member.items():
+            member_representation = self._create_member_representation(member_name)
+            member_representation["children"] = assets
+            members_grid.append(member_representation)
+
+        return {
+            "name": "Grid Market",
```

Comment:
AFAIS from the implementation of a SCM [setup in Spyros' PR](https://github.com/gridsingularity/gsy-e/pull/1454/files#diff-2fbe96ece8e24e666bfea3890b8aa5ddcf9e16f66a91ba121c8de6384ed6af84R33), I think there will be no grid market, only the community. 
```suggestion
            "name": "Community",
```
Consequently, it also makes sense to rename the method.
Or did you want to be future prove when we also wan to enable adding communities (like in the new UI sense) with the datasheet? If the latter is true, we need to add another hierarchy layer including a market maker.

Reply:
Thanks, that's exactly what I was looking for. I wasn't sure about how the grid should have been structured, so I just looked into our current grid representations and copied them.

That's also why in the retro I was asking to create schemas for validation: I think we need to document how grids should be built and what's valid/invalid.

Reply:
I think that specifically for SCM it is up to us and not up to the frontend. Since the only way to create an SCM community is via the community datasheet, we have the liberty to use any internal representation we want. And, in order to hopefully facilitate things hopefully for everyone, I took the easiest way out: remove the grid market, throw away the market maker area/strategy, and keep only the community as a root-level area. 

WRT validation of the grid structure, I was under the impression on our discussion that it was mostly about validation of the parameters and not of the grid structure. If it was about the GSy maps, I would not agree with you because they can in theory not be used only with the 4-level hierarchy that we currently have. However for the SCM you are probably right that a proper validation would save us from some bugs along the way. However I do not think it is very urgent, and can even be offloaded to a 'hackathon' ticket, since it is bite-size and can probably be completed in a couple of hours.

Reply:
> WRT validation of the grid structure, I was under the impression on our discussion that it was mostly about validation of the parameters and not of the grid structure

I think the two can come together: if we use the JSON schema validator, we can check for parameters types but also structure. Schemas can also be inherited/composed, so some parts of them can be shared (and we can have one schema for the SCM structure and another one for the older hierarchy that reuse the same sub-schemas).

In general, I think a possible DoD might be something like:
- always validate the data we pass from service X to service Y (e.g. gsy-web <-> gsy-e, or datasheet -> gsy-web);
- execute validation in just a few key locations (without having too many validation methods scattered around the codebase);
- have a well-defined, exhaustive schema that can be used as documentation (this is what I missed while working on this feature).